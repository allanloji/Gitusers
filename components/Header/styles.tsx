import styled from '@emotion/styled';

const Container = styled.div`
  width: 100%;
  height: 400px;
  background: #8ecae6;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
`;

const Title = styled.h1`
  font-size: 120px;
  font-weight: bold;
  margin: 0.5rem;
  @media (max-width: 768px) {
    font-size: 80px;
  }
`;

const Emoji = styled.span`
  font-size: 100px;
`;

const Subtitle = styled.h2`
  font-size: 30px;
`;

export { Container, Title, Emoji, Subtitle };
