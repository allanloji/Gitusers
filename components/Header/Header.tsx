import React from 'react';

import { Container, Emoji, Subtitle, Title } from './styles';

/**
 * Header
 */
function Header() {
  return (
    <Container>
      <Title>gitusers </Title>
      <Emoji>👦🏻</Emoji>
      <Subtitle>A github user search</Subtitle>
    </Container>
  );
}

export default Header;
