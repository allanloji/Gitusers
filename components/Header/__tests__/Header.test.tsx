import React from 'react';
import { render, screen } from '@testing-library/react';

import Header from '..';

describe('Header', () => {
  test('renders successfully', () => {
    render(<Header />);
    const element = screen.getByTestId(/test/i);
    expect(element).toBeInTheDocument();
  });
});
