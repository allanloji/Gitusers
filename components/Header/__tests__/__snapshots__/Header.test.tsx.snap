// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render 1`] = `
<div>
  <div
    class="css-1op182y"
  >
    <h1
      class="css-r838ov"
    >
      gitusers 
    </h1>
    <span
      class="css-dkt9vh"
    >
      👦🏻
    </span>
    <h2
      class="css-vkg5qd"
    >
      A github user search
    </h2>
  </div>
</div>
`;
